### ISSUE TITLE
Files ignored on push directly after clone

### ISSUE THREAD

**Issue by markjwiggins** [google/clasp#669]
## Expected Behavior
After cloning a script, any subfolder files (e.g. gs/Code.gs) are ignored when subsequently running `clasp status` and ignored and deleted from the script file when running `clasp push`.

## Actual Behavior
Subfolder files are ignored.
```
user@localhost:~/Downloads/clasp/clasptest$ sudo clasp clone "xxxxx"
Cloned 3 files.
└─ gs/inner.js
└─ appsscript.json
└─ Code.js
user@localhost:~/Downloads/clasp/clasptest$ sudo clasp status
Not ignored files:
└─ Code.js
└─ appsscript.json

Ignored files:
└─ .clasp.json
└─ gs/inner.js
```

## Steps to Reproduce the Problem

1. Clone script
2. Run `clasp status` or `clasp push` = Subfolder files ignored

## Specifications

- Node version (`node -v`): 8.16.0
- Version (`clasp -v`): 2.2.1
- OS (Mac/Linux/Windows): Ubuntu 16.04 running on ChromeOS Crouton

**Comment by PopGoesTheWza** [comment]
@grant I guess either the *default* `.claspignore` should include files in subdirectory or documentation needs to be changed so that end users are aware that after a `clasp clone` some additional configuration (i.e. `.claspignore` setup) may be necessary. Personally, I'd opt for the second option.

In any cases, it could be valuable to the end user that a `clasp status` like report be issued after a successful `clasp clone`, IMHO.

Please share your insight on those design consideration.

**Comment by markjwiggins** [comment]
@PopGoesTheWza Thank you, based on your reply I looked through the `.claspignore` text and the key sentence was
> The default will only consider the appsscript.json manifest and any JavaScript and TypeScript source files in the `rootDir` folder

So creating the `.claspignore` file and populating it to include subdirectories looks to solve my issue.
So the application is working as expected, but I would agree making this item clear after a `clasp clone` would be helpful.

**Comment by MattiasMartens** [comment]
Was about to create an issue for this. The inclusion of subdirectories in the default ignore is a breaking change. My existing projects, which had already been cloned, were invalidated by this update.

This is a bit of a sob story, but the effects of this were pretty bad. I'm maintaining an app for a client. I pushed a minor change to production, having already checked it on my staging app. In between I updated Clasp because something wasn't working. This introduced the default ignore list. The new version of Clasp then pushed only the root files, ignoring all of the files in the subdirectories (essentially the entire app).

This resulted in a series of very concerned messages on Saturday morning--they work weekends and are up before I am. Reverting was easy, but it then took a couple hours to determine what the hell happened.

It's on me for not verifying that the app was working before I went to bed, of course, but I still feel that the principle of least astonishment was violated here.

The exclusion of subfolders by default is especially confusing because of these lines in the documentation:
```
Structure Code: clasp automatically converts your flat project on script.google.com into folders.
```

I suppose this is technically true, but the reverse is not true -- Clasp no longer converts your nested structure into a flat project on script.google.com, at least not without configuration.

@PopGoesTheWza, you prefer ignoring subdirectories, correct? Can I ask why? I could see if it was to ignore `node_modules`, but... why not just ignore `node_modules/**` by default instead of ignoring everything?

I know it all comes down to preference and I will of course add empty `.claspignore` files to all my projects now. I just want to register with y'all that I really, really don't like it.
